consensus/istanbul/backend/handler.go

```diff
@@ -286,3 +301,175 @@ func (sb *Backend) UnregisterPeer(peer consensus.Peer, isProxiedPeer bool) {
 		}
 	}
 }
+
+// Handshake allows the initiating peer to identify itself as a validator
+func (sb *Backend) Handshake(peer consensus.Peer) (bool, error) {
+	// Only written to if there was a non-nil error when sending or receiving
+	errCh := make(chan error, 1)
+	isValidatorCh := make(chan bool, 1)
+
+	sendHandshake := func() {
+		var validatorProofMsg *istanbul.Message
+		var err error
+		containsProof := false
+		if peer.PurposeIsSet(p2p.ValidatorPurpose) {
+			// validatorProofMsg may be nil
+			validatorProofMsg, err = sb.retrieveVersionedEnodeMsg()
+			if err != nil {
+				errCh <- err
+				return
+			}
+		}
+		// Even if we decide not to identify ourselves,
+		// send an empty message to complete the handshake
+		if validatorProofMsg == nil {
+			validatorProofMsg = &istanbul.Message{}
+		} else {
+			containsProof = true
+		}
+		msgBytes, err := validatorProofMsg.Payload()
+		if err != nil {
+			errCh <- err
+			return
+		}
+		err = peer.Send(istanbulValidatorProofMsg, msgBytes)
+		if err != nil {
+			errCh <- err
+		}
+		isValidatorCh <- containsProof
```

Comment:
If the peer is validatorpurpose it will automatically bypass the max peer check, but changing this to be `peer.PurposeIsSet(p2p.ValidatorPurpose)` is much more clear. Fixed!